### PR TITLE
system-variables: add `tidb_opt_fix_control`

### PR DESCRIPTION
**Pull request by time-and-fate**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
https://github.com/pingcap/tidb/issues/43169

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: system-variables.md [hunk 1]
```diff
@@ -2597,6 +2597,16 @@ mysql> desc select count(distinct a) from test.t;
...
 - 默认情况下，如果 `SELECT` 语句中包含过滤条件（`WHERE` 子句），TiFlash 会先扫描查询所需列的全部数据后再进行过滤。当设置该变量为 `ON` 开启 TiFlash 延迟物化功能时，TiFlash 可以先扫描过滤条件相关的列数据，过滤得到符合条件的行后，再扫描这些行的其他列数据，继续后续计算，从而减少 IO 扫描和数据处理的计算量。
 
+### `tidb_opt_fix_control` <span class="version-mark">从 v7.1.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：字符串
+- 默认值：`""`
+- 这个变量用来控制优化器的一些内部行为。
+- 为了能让用户能更细粒度地控制优化器的行为，并且有能力防止优化器行为变化导致的升级回退，优化器的一部分行为细节将通过此系统变量暴露和控制。
```
Comment (songrijie) [inline system-variables.md:2607]:
```suggestion
- 一部分优化器行为的选择依赖用户场景或 SQL 的编写。 为了让用户能更细粒度地控制优化器的行为，并且有能力防止优化器行为变化导致的性能回退，通过此系统变量进行干预。
```

File: system-variables.md [hunk 2]
```diff
@@ -2597,6 +2597,16 @@ mysql> desc select count(distinct a) from test.t;
...
+- 每个控制项会对应一个 GitHub Issue，可以通过该变量设置为一个值。设置多个控制项时，不同控制项之间使用逗号隔开。格式形如 `"<#issue1>:<value1>,<#issue2>:<value2>"`，其中 `<#issueN>` 表示 GitHub Issue 编号，`<valueN>` 表示希望设置的值。
```
Comment (songrijie) [inline system-variables.md:2608]:
```suggestion
- 该变量支持设置多个控制项，不同控制项之间使用逗号隔开。格式形如 `"<#issue1>:<value1>,<#issue2>:<value2>"`，其中 `<#issueN>` 代表控制项编号，对应链接 `https://github.com/pingcap/tidb/issues/<#issue>` 对该控制项进行说明，`<valueN>` 是为该控制项设置的目标值。
```

**Review by hfxsd — APPROVED** [review]

File: system-variables.md [hunk 3]
```diff
@@ -2597,6 +2597,16 @@ mysql> desc select count(distinct a) from test.t;
...
 
+### `tidb_opt_fix_control` <span class="version-mark">从 v7.1.0 版本开始引入</span>
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 类型：字符串
+- 默认值：`""`
+- 这个变量用来控制优化器的一些内部行为。
+- 一部分优化器行为的选择依赖用户场景或 SQL 的编写。该变量可以让你更细粒度地控制优化器的行为，并且可以防止优化器行为变化导致的性能回退。
+- 该变量支持设置多个控制项，不同控制项之间使用逗号隔开。格式形如 `"<#issue1>:<value1>,<#issue2>:<value2>"`，其中 `<#issueN>` 代表控制项编号，对应链接 `https://github.com/pingcap/tidb/issues/<#issue>` 对该控制项进行说明，`<valueN>` 是为该控制项设置的目标值。
```
Comment (Oreoxmt) [inline system-variables.md:2608]:
```suggestion
- 一部分优化器行为的选择依赖用户场景或 SQL 编写方式。通过设置该变量，你可以更细粒度地控制优化器的行为，并且避免优化器行为变化导致的性能回退。
- 该变量支持设置多个控制项，不同控制项之间使用逗号 (`,`) 分隔。格式形如 `"<#issue1>:<value1>,<#issue2>:<value2>"`，其中 `<#issueN>` 代表控制项编号，对应链接 `https://github.com/pingcap/tidb/issues/<#issue>` 对该控制项进行说明，`<valueN>` 是为该控制项设置的目标值。
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Review by chrysan — APPROVED** [review]

File: control-execution-plan.md [hunk 4]
```diff
@@ -10,3 +10,4 @@ SQL 性能调优的前两个章节介绍了如何理解 TiDB 的执行计划以
 - [Optimizer Hints](/optimizer-hints.md)中，我们会介绍如何使用 Hint 来指导 TiDB 生成执行计划。
 - 但是使用 Hint 会侵入性地更改 SQL，在一些场景下并不能简单的插入 Hint。在[执行计划管理](/sql-plan-management.md)中，我们会介绍 TiDB 如何使用另一种语法来非侵入地控制执行计划的生成，同时还会介绍后台自动对执行计划进行演进的手段。该手段可用来减轻诸如版本升级等原因造成的执行计划不稳定和集群性能下降的问题。
 - 最后在[优化规则及表达式下推的黑名单](/blocklist-control-plan.md)中，我们会介绍黑名单的使用。
+- 除以上手段之外，执行计划还会被一部分[系统变量](/system-variables.md)所影响，通过在系统级或会话级对变量进行修改，能够控制执行计划的生成。 自 v7.1.0 起，我们引入了一个相对特殊的变量 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v710-版本开始引入)， 这个变量能够接受多个控制项，用来更细粒度地控制优化器的行为，避免集群升级后优化器行为变化导致的性能回退。
```
Comment (hfxsd) [inline control-execution-plan.md:13]:
```suggestion
- 除以上手段之外，执行计划还会被一部分[系统变量](/system-variables.md)所影响，通过在系统级或会话级对变量进行修改，能够控制执行计划的生成。 自 v7.1.0 起，TiDB 引入了一个相对特殊的变量 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v710-版本开始引入)， 这个变量能够接受多个控制项，用来更细粒度地控制优化器的行为，避免集群升级后优化器行为变化导致的性能回退。
```

File: control-execution-plan.md [hunk 5]
```diff
@@ -10,3 +10,4 @@ SQL 性能调优的前两个章节介绍了如何理解 TiDB 的执行计划以
 - [Optimizer Hints](/optimizer-hints.md)中，我们会介绍如何使用 Hint 来指导 TiDB 生成执行计划。
 - 但是使用 Hint 会侵入性地更改 SQL，在一些场景下并不能简单的插入 Hint。在[执行计划管理](/sql-plan-management.md)中，我们会介绍 TiDB 如何使用另一种语法来非侵入地控制执行计划的生成，同时还会介绍后台自动对执行计划进行演进的手段。该手段可用来减轻诸如版本升级等原因造成的执行计划不稳定和集群性能下降的问题。
 - 最后在[优化规则及表达式下推的黑名单](/blocklist-control-plan.md)中，我们会介绍黑名单的使用。
+- 除以上手段之外，执行计划还会被一部分[系统变量](/system-variables.md)所影响，通过在系统级或会话级对变量进行修改，能够控制执行计划的生成。 自 v7.1.0 起，TiDB 引入了一个相对特殊的变量 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v710-版本开始引入)， 这个变量能够接受多个控制项，用来更细粒度地控制优化器的行为，避免集群升级后优化器行为变化导致的性能回退。
```
Comment (Oreoxmt) [inline control-execution-plan.md:13]:
```suggestion
- 除以上手段之外，执行计划还会被一部分[系统变量](/system-variables.md)所影响，通过在系统级或会话级对变量进行修改，能够控制执行计划的生成。自 v7.1.0 起，TiDB 引入了一个相对特殊的变量 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v710-版本开始引入)。这个变量能够接受多个控制项，用来更细粒度地控制优化器的行为，避免集群升级后优化器行为变化导致的性能回退。
```

File: control-execution-plan.md [hunk 6]
```diff
@@ -10,3 +10,4 @@ SQL 性能调优的前两个章节介绍了如何理解 TiDB 的执行计划以
 - [Optimizer Hints](/optimizer-hints.md)中，我们会介绍如何使用 Hint 来指导 TiDB 生成执行计划。
 - 但是使用 Hint 会侵入性地更改 SQL，在一些场景下并不能简单的插入 Hint。在[执行计划管理](/sql-plan-management.md)中，我们会介绍 TiDB 如何使用另一种语法来非侵入地控制执行计划的生成，同时还会介绍后台自动对执行计划进行演进的手段。该手段可用来减轻诸如版本升级等原因造成的执行计划不稳定和集群性能下降的问题。
 - 最后在[优化规则及表达式下推的黑名单](/blocklist-control-plan.md)中，我们会介绍黑名单的使用。
+- 除以上手段之外，执行计划还会被一部分[系统变量](/system-variables.md)所影响，通过在系统级或会话级对变量进行修改，能够控制执行计划的生成。自 v7.1.0 起，TiDB 引入了一个相对特殊的变量 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v710-版本开始引入)。这个变量能够接受多个控制项，用来更细粒度地控制优化器的行为，避免集群升级后优化器行为变化导致的性能回退。
```
Comment (hfxsd) [inline control-execution-plan.md:13]:
```suggestion
- 除以上手段之外，执行计划还会被一部分系统变量所影响，通过在系统级或会话级对变量进行修改，能够控制执行计划的生成。自 v7.1.0 起，TiDB 引入了一个相对特殊的变量 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v710-版本开始引入)。这个变量能够接受多个控制项，用来更细粒度地控制优化器的行为，避免集群升级后优化器行为变化导致的性能回退。
```

File: control-execution-plan.md [hunk 7]
```diff
@@ -10,3 +10,4 @@ SQL 性能调优的前两个章节介绍了如何理解 TiDB 的执行计划以
 - [Optimizer Hints](/optimizer-hints.md)中，我们会介绍如何使用 Hint 来指导 TiDB 生成执行计划。
 - 但是使用 Hint 会侵入性地更改 SQL，在一些场景下并不能简单的插入 Hint。在[执行计划管理](/sql-plan-management.md)中，我们会介绍 TiDB 如何使用另一种语法来非侵入地控制执行计划的生成，同时还会介绍后台自动对执行计划进行演进的手段。该手段可用来减轻诸如版本升级等原因造成的执行计划不稳定和集群性能下降的问题。
 - 最后在[优化规则及表达式下推的黑名单](/blocklist-control-plan.md)中，我们会介绍黑名单的使用。
+- 除以上手段之外，执行计划还会被一部分系统变量所影响，通过在系统级或会话级对变量进行修改，能够控制执行计划的生成。自 v7.1.0 起，TiDB 引入了一个相对特殊的变量 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v710-版本开始引入)。这个变量能够接受多个控制项，用来更细粒度地控制优化器的行为，避免集群升级后优化器行为变化导致的性能回退。
```
Comment (hfxsd) [inline control-execution-plan.md:13]:
```suggestion

除以上手段之外，执行计划还会被一部分系统变量所影响，通过在系统级或会话级对变量进行修改，能够控制执行计划的生成。自 v7.1.0 起，TiDB 引入了一个相对特殊的变量 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v710-版本开始引入)。这个变量能够接受多个控制项，用来更细粒度地控制优化器的行为，避免集群升级后优化器行为变化导致的性能回退。
```

File: control-execution-plan.md [hunk 8]
```diff
@@ -10,3 +10,5 @@ SQL 性能调优的前两个章节介绍了如何理解 TiDB 的执行计划以
 - [Optimizer Hints](/optimizer-hints.md)中，我们会介绍如何使用 Hint 来指导 TiDB 生成执行计划。
 - 但是使用 Hint 会侵入性地更改 SQL，在一些场景下并不能简单的插入 Hint。在[执行计划管理](/sql-plan-management.md)中，我们会介绍 TiDB 如何使用另一种语法来非侵入地控制执行计划的生成，同时还会介绍后台自动对执行计划进行演进的手段。该手段可用来减轻诸如版本升级等原因造成的执行计划不稳定和集群性能下降的问题。
 - 最后在[优化规则及表达式下推的黑名单](/blocklist-control-plan.md)中，我们会介绍黑名单的使用。
+
+除以上手段之外，执行计划还会被一部分系统变量所影响，通过在系统级或会话级对变量进行修改，能够控制执行计划的生成。自 v7.1.0 起，TiDB 引入了一个相对特殊的变量 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v710-版本开始引入)。这个变量能够接受多个控制项，用来更细粒度地控制优化器的行为，避免集群升级后优化器行为变化导致的性能回退。
```
Comment (Oreoxmt) [inline control-execution-plan.md:14]:
```suggestion
除以上手段之外，执行计划还会被一部分系统变量所影响。通过在系统级或会话级对变量进行修改，能够控制执行计划的生成。自 v7.1.0 起，TiDB 引入了一个相对特殊的变量 [`tidb_opt_fix_control`](/system-variables.md#tidb_opt_fix_control-从-v710-版本开始引入)。这个变量能够接受多个控制项，用来更细粒度地控制优化器的行为，避免集群升级后优化器行为变化导致的性能回退。
```

**Review by Oreoxmt — APPROVED** [review]

**Pull request merged**